### PR TITLE
[ENT] Versioned Migration via Atlas

### PR DESCRIPTION
**Pull request by pxp928**
# Description of the PR

This adds the ability to run versioned migration for ENT whenever the schema changes. 

This follows the recommendation provided in: https://entgo.io/docs/versioned-migrations/#quick-guide

Checks have been added to the CI to check that the migration diff has been generated and a linter has been run via atlas.

**Note:** For production use cases auto migration should be disabled and atlas should be used to apply the initial migration!!

closes  https://github.com/guacsec/guac/issues/1513

Example usage:

1. make changes to the schema (for example add a new optional field in artifact)
2. run `make generate` which will also run atlas (needs to have docker running). This will generate the migration plan in `pkg/assembler/backends/ent/migrate/migrations`
3. Run
```
atlas migrate apply \
  --dir "file://pkg/assembler/backends/ent/migrate/migrations" \
  --url "postgres://guac:guac@0.0.0.0:5432/guac?search_path=public&sslmode=disable" \
````
`--url` must be set to the running production database
`--baseline` specified the baseline that is currently applied to the running database (if started without atlas)

Note: A new baseline can be created from the production server by running:
```
atlas migrate diff my_baseline \
  --dir "file://pkg/assembler/backends/ent/migrate/migrations" \
  --dev-url "docker://postgres/15/test?search_path=public" \
  --to "postgres://guac:guac@0.0.0.0:5432/guac?search_path=public&sslmode=disable"
```

After successful execution, you will see the following message:
```
  Migrating to version 20240503144258 from 20240503123155 (1 migrations in total):

    -- migrating version 20240503144258
      -> ALTER TABLE "artifacts" ADD COLUMN "test" character varying NULL;
    -- ok (4.024333ms)

    -------------------------
    -- 12.905875ms
    -- 1 migration
    -- 1 sql statement
```

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]
Thank you!

File: pkg/assembler/backends/ent/schema/certifyvex.go [hunk 1]
```diff
@@ -62,7 +62,9 @@ func (CertifyVex) Edges() []ent.Edge {
 // Indexes of the VEX.
 func (CertifyVex) Indexes() []ent.Index {
 	return []ent.Index{
-		index.Fields("known_since", "justification", "status", "statement", "status_notes", "origin", "collector", "document_ref").Edges("vulnerability", "package").Unique().Annotations(entsql.IndexWhere("artifact_id IS NULL")),
-		index.Fields("known_since", "justification", "status", "statement", "status_notes", "origin", "collector", "document_ref").Edges("vulnerability", "artifact").Unique().Annotations(entsql.IndexWhere("package_id IS NULL")),
+		index.Fields("known_since", "justification", "status", "statement", "status_notes", "origin", "collector", "document_ref").
+			Edges("vulnerability", "package").Unique().Annotations(entsql.IndexWhere("artifact_id IS NULL")).StorageKey("vex_artifact_id"),
```
Comment (pxp928) [inline pkg/assembler/backends/ent/schema/certifyvex.go:66]: `StorageKey` was needed here (and the other locations) as the atlas was getting confused and thought it was the same index twice.
